vassal-app/src/main/java/VASSAL/script/ExpressionInterpreter.java

```diff
@@ -231,70 +246,80 @@ public String evaluate(PropertySource ps, boolean localized) throws ExpressionEx
       return "";
     }
 
-    // Default to the GameModule to satisfy properties if no
-    // GamePiece supplied.
-    source = ps == null ? GameModule.getGameModule() : ps;
-
-    setNameSpace(expressionNameSpace);
-
-    // Bind each undeclared variable with the value of the
-    // corresponding Vassal property. Allow for old-style $variable$ references
-    for (String var : variables) {
-      String name = var;
-      if (name.length() > 2 && name.startsWith("$") && name.endsWith("$")) {
-        name = name.substring(1, name.length() - 1);
-      }
-      Object prop = localized ? source.getLocalizedProperty(name) : source.getProperty(name);
-      String value = prop == null ? "" : prop.toString();
-      if (value == null) {
-        setVar(var, "");
-      }
-      else if ("true".equals(value)) {
-        setVar(var, true);
-      }
-      else if ("false".equals(value)) {
-        setVar(var, false);
-      }
-      else {
-        try {
-          setVar(var, Integer.valueOf(value).intValue());
+    String result;
+    try {
+      RecursionLimiter.startExecution(this);
+
+      // Default to the GameModule to satisfy properties if no
+      // GamePiece supplied.
+      source = ps == null ? GameModule.getGameModule() : ps;
+  
+      setNameSpace(expressionNameSpace);
+  
+      // Bind each undeclared variable with the value of the
+      // corresponding Vassal property. Allow for old-style $variable$ references
+      for (String var : variables) {
+        String name = var;
+        if (name.length() > 2 && name.startsWith("$") && name.endsWith("$")) {
+          name = name.substring(1, name.length() - 1);
         }
-        catch (NumberFormatException e) {
+        Object prop = localized ? source.getLocalizedProperty(name) : source.getProperty(name);
+        String value = prop == null ? "" : prop.toString();
+        if (value == null) {
+          setVar(var, "");
+        }
+        else if ("true".equals(value)) {
+          setVar(var, true);
+        }
+        else if ("false".equals(value)) {
+          setVar(var, false);
+        }
+        else {
           try {
-            setVar(var, Float.valueOf(value).floatValue());
+            setVar(var, Integer.valueOf(value).intValue());
```

Comment:
Boxing/unboxing to parse a primitive VASSAL.script.ExpressionInterpreter.evaluate(PropertySource, boolean) [VASSAL.script.ExpressionInterpreter] At ExpressionInterpreter.java:[line 279] DM_BOXED_PRIMITIVE_FOR_PARSING (https://travis-ci.com/github/vassalengine/vassal/builds/179361630#L544, http://findbugs.sourceforge.net/bugDescriptions.html#DM_BOXED_PRIMITIVE_FOR_PARSING)

Reply:
`Integer.valueOf(value).intValue()` can just be `Integer.parseInt(value)` (and similarly, `Float.parseFloat(value)`).